### PR TITLE
Increase timeout for object manager valgrind tests

### LINKED ISSUE #3979: Valgrind object manager test failure.
I saw this test failure in valgrind recently, see e.g., https://travis-ci.com/ray-project/ray/jobs/176127234
```
bash ../src/ray/test/run_object_manager_valgrind.sh
```

The first time I saw this was today in the master branch build for https://github.com/ray-project/ray/pull/3756.

```
$ bash ../src/ray/test/run_object_manager_valgrind.sh
++dirname ../src/ray/test/run_object_manager_valgrind.sh
+SCRIPT_DIR=../src/ray/test
+RAY_ROOT=../src/ray/test/../../..
++cd ../src/ray/test/../../..
++pwd
+RAY_ROOT=/home/travis/build/ray-project/ray
+'[' -z /home/travis/build/ray-project/ray ']'
+'[' '!' -d /home/travis/build/ray-project/ray/python ']'
+CORE_DIR=/home/travis/build/ray-project/ray/build
+PYTHON_CORE_DIR=/home/travis/build/ray-project/ray/python/ray/core
+REDIS_DIR=/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src
+REDIS_MODULE=/home/travis/build/ray-project/ray/python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so
+STORE_EXEC=/home/travis/build/ray-project/ray/python/ray/core/src/plasma/plasma_store_server
+VALGRIND_CMD='valgrind --track-origins=yes --leak-check=full --show-leak-kinds=all --leak-check-heuristics=stdstring --error-exitcode=1'
+[[ '' = \o\n ]]
+REDIS_SERVER=/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-server
+LOAD_MODULE_ARGS='--loadmodule /home/travis/build/ray-project/ray/python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so'
+echo /home/travis/build/ray-project/ray/python/ray/core/src/plasma/plasma_store_server
/home/travis/build/ray-project/ray/python/ray/core/src/plasma/plasma_store_server
+echo '/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-server --loglevel warning --loadmodule /home/travis/build/ray-project/ray/python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so --port 6379'
/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-server --loglevel warning --loadmodule /home/travis/build/ray-project/ray/python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so --port 6379
+echo '/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-cli -p 6379 shutdown'
/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-cli -p 6379 shutdown
+killall plasma_store
plasma_store: no process found
+true
+/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-cli -p 6379 shutdown
Could not connect to Redis at 127.0.0.1:6379: Connection refused
+true
+sleep 1s
+sleep 1s
+/home/travis/build/ray-project/ray/build/src/ray/thirdparty/redis/src/redis-server --loglevel warning --loadmodule /home/travis/build/ray-project/ray/python/ray/core/src/ray/gcs/redis_module/libray_redis_module.so --port 6379
24938:C 07 Feb 2019 19:15:17.208 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
24938:C 07 Feb 2019 19:15:17.208 # Redis version=5.0.3, bits=64, commit=f9875727, modified=0, pid=24938, just started
24938:C 07 Feb 2019 19:15:17.208 # Configuration loaded
24938:M 07 Feb 2019 19:15:17.209 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
24938:M 07 Feb 2019 19:15:17.209 # Server initialized
24938:M 07 Feb 2019 19:15:17.209 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
24938:M 07 Feb 2019 19:15:17.209 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
+valgrind --track-origins=yes --leak-check=full --show-leak-kinds=all --leak-check-heuristics=stdstring --error-exitcode=1 /home/travis/build/ray-project/ray/build/src/ray/object_manager/object_manager_test /home/travis/build/ray-project/ray/python/ray/core/src/plasma/plasma_store_server
==24943== Memcheck, a memory error detector
==24943== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==24943== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==24943== Command: /home/travis/build/ray-project/ray/build/src/ray/object_manager/object_manager_test /home/travis/build/ray-project/ray/python/ray/core/src/plasma/plasma_store_server
==24943== 
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestObjectManager
[ RUN      ] TestObjectManager.StartTestObjectManager
==24943== Warning: set address range perms: large range [0x3a044000, 0x75a04000) (defined)
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0207 19:15:27.243613 24943 object_manager.cc:349] Invalid Push request ObjectID: 8ae7703d66b770db210a9f5ca43a3d9f0f43b901 after waiting for 1000 ms.
==24943== Warning: set address range perms: large range [0x75a04000, 0xb13c4000) (defined)
==24943== Warning: set address range perms: large range [0xb13c4000, 0xecd84000) (defined)
==24943== Warning: set address range perms: large range [0xecd84000, 0x128744000) (defined)
==24943== Warning: set address range perms: large range [0x128744000, 0x164104000) (defined)
==24943== Warning: set address range perms: large range [0x164104000, 0x19fac4000) (defined)
/home/travis/build/ray-project/ray/src/ray/object_manager/test/object_manager_test.cc:452: Failure
Value of: found.size() >= required_objects
  Actual: false
Expected: true
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

cc @stephanie-wang

### PR DESCRIPTION
**Pull request by stephanie-wang**
## What do these changes do?

Since the inlined objects PR (#3756) was merged, one of the object manager tests was failing on valgrind due to a timing issue. This increases the timeout when running in valgrind.

This also removes an unnecessary data copy when retrieving the inline object data.

## Related issue number

Closes #3979.

### CODE REVIEW

File: src/ray/object_manager/test/object_manager_test.cc [hunk 1]
```diff
@@ -342,27 +345,27 @@ class TestObjectManager : public TestObjectManagerBase {
     case 0: {
       // Ensure timeout_ms = 0 is handled correctly.
       // Out of 5 objects, we expect 3 ready objects and 2 remaining objects.
-      TestWait(600, 5, 3, /*timeout_ms=*/0, false, false);
+      TestWait(100, 5, 3, /*timeout_ms=*/0, false, false);
```
Comment (pcmoritz) [inline src/ray/object_manager/test/object_manager_test.cc]: Is decreasing the object size intentional here?

Comment (stephanie-wang) [reply src/ray/object_manager/test/object_manager_test.cc]: Yes, to put it back under the inline objects limit.

Comment (pcmoritz) [reply src/ray/object_manager/test/object_manager_test.cc]: I see, shouldn't we test both?

File: src/ray/test/run_object_manager_tests.sh [hunk 2]
```diff
@@ -45,7 +45,10 @@ sleep 1s
 # Run tests.
 $CORE_DIR/src/ray/object_manager/object_manager_stress_test $STORE_EXEC
 sleep 1s
-$CORE_DIR/src/ray/object_manager/object_manager_test $STORE_EXEC
+# Use timeout=1000ms for the Wait tests.
+$CORE_DIR/src/ray/object_manager/object_manager_test $STORE_EXEC 1000
+# Run tests again with inlined objects.
+$CORE_DIR/src/ray/object_manager/object_manager_test $STORE_EXEC 1000
```
Comment (guoyuhong) [inline src/ray/test/run_object_manager_tests.sh]: Why use the same parameters for the same test? Is `true` missing?

**Review by guoyuhong — DISMISSED** [review]
Thanks for fix this! I left a minor comment.

File: src/ray/test/run_object_manager_tests.sh (hunk 2 above)
Comment (stephanie-wang) [reply src/ray/test/run_object_manager_tests.sh]: Ah, thank you! I added it.

**Comment by pcmoritz** [comment]
Jenkins retest this please

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**